### ISSUE TITLE
Need an easy way to prevent screen lock

### ISSUE THREAD

**Issue by mwlang** [Jasonette/JASONETTE-iOS#146]
Similar to how Waze and other apps can prevent a screen lock, Jasonette should support toggling the screen lock of devices.

http://stackoverflow.com/questions/28329185/how-to-prevent-screen-lock-on-my-application-with-swift-and-ios-8

https://developer.apple.com/reference/uikit/uiapplication#1657524

**Comment by clsource** [comment]
this could be made into a extension 👍

**Comment by clsource** [comment]
Done
https://github.com/NinjasCL/JasonIdleTimerAction

**Comment by mwlang** [comment]
@clsource awesome!  I think I just learned a new trick from you as well.  I do think something like this would be very good addition to the core, though.

**Comment by gliechtenstein** [comment]
Yeah I think we could incorporate this into core. This is why I made Jasonette extensible, so we can experiment simply by writing extensions, and if enough people want it and it makes sense to incorporate into core we can easily do so.

Also incorporating to core means we need to support Android as well (which is probably possible as well). 

Let's update this thread when there's any update on this in the future.

**Comment by mcken2ja** [comment]
This is a pretty cool feature, even if it only supports iOS right now. Something I noticed about the sample code as it seems the files should be JasonIdleAction based on the documentation of how to add in a new action. If you make that slight change something like this would work:
"actions": {
        "$load": {
          "type": "$idle.disable"
        },
}

**Comment by clsource** [comment]
@mcken2ja that's already possible since the caller name is defined here

https://github.com/NinjasCL-jasonette/JasonIdleTimerAction/blob/master/jr.json

as `idle`

👍
